### ISSUE TITLE
[NPC] [Shadow Labyrinth] - Cabal Assassin Patrol

### ISSUE THREAD

**Issue by Drayo** [Atlantiss/NetherwingBugtracker#7052]
**Description**: The Assassins in SL are not patrolling properly and there is only one of them present.

**Current behaviour**:  

- There seems to be only one Cabal Assassin in SL (pre-spawned) in the hallway leading to the first Boss.

- They do not seem to use Cheapshot.

**Expected behaviour**: 

- The **assassins should spawn periodically after the first Boss dies and until the second Boss is killed.** The maximum that can be up at any given time seems to be two. They should patrol according to the images. Apologies for the paint action, but a video is also shit because they move so slowly. The path drawn is like 90% correct, idk where exactly on the path they spawn so I guessed.

1. https://i.imgur.com/xCd9QIU.jpg
Starting on the Black Spot, one round through the first Boss room, then a loop on the other side, then, drawn in yellow, a second loop through the boss room and back to the start.

2.  https://i.imgur.com/8nZCyhn.jpg
This guy is already spawned on NW, he should not spawn until Hellmaw is dead. He runs a "perfect loop" unlike the first one, he never turns 180° so I cannot tell where exactly he spawned, but back in the Hallway where the current one is pre-spawned might make the most sense.

Small Note regarding the paths, the lines should overlap, they always go through the middle of the connector, but that is kinda hard to draw so I did it this way. You know what I mean.

- **They should Cheapshot the first target they attack out of stealth.**

**Source**: Retail/Wowhead/wiki

- https://www.wowhead.com/npc=18636/cabal-assassin#comments

 By chonchu (7,457 – 2·24·57) on 2007/02/07 (Patch 2.0.6)
> These guys are stealthed rogues.
> They spawn once you kill the first boss, they pat the first boss room and the 2nd boss room.
> I don´t know much info about them like where they spawn or where they come from or even if they are linked with a timer or something.
> The only thing that happens is: we´re fightin a pull and they just sap someone, cheap shot someone with poisons and they´re very strong.

By 15939 (646 –  1·5)  on 2007/02/25 (Patch 2.0.8) |   |  
> I'm reasonably certain that they're not invisible, or stealthed, but that they randomly spawn on you.

By Raatu (859 –  7·9)  on 2007/02/27 (Patch 2.0.8)
> As of Patch 2.0.10, these assassin attacks will be less frequent!  Yipee!  http://www.wow-europe.com/en/info/underdev/testrealm.html (accurate as of 27/02/07)

https://worldofwarcraft.fandom.com/et/wiki/Shadow_Labyrinth
> Spawns during the pulls prior to Blackheart the Inciter.  Assassins appear to spawn periodically; the > longer you spend in the  entry to Blackheart's room, the more times you will see them spawn.   > There are reports of them spawning two at a time on occasion.

This implies that they are not spawned until the first Boss is dead, and then they will respawn over and over until Blackheart is dead. A maximum of two can be up at the same time. In 2.0 there should be more frequent spawns.

**Server Revision**: 3396

**Comment by Tereneckla** [comment]
https://youtu.be/9aLlWZdHa74?t=905
This video from retail (but not tbc) has an Assassin pop up on the left side of the 2nd boss room.
Uses cheap shot and as the DK stood still for a while patrols.

Would be nice to know where you got those patrol paths from.

**Comment by Drayo** [comment]
I got the path from retail. Once the first boss died I just waited a few minutes for them to spawn, then followed each of them for one cycle. They are super easy to spot with a high level after all.

A note on spawn times. Sometimes it took 3 minutes for even one of them to spawn, sometimes both of them spawned pretty much instantly. I have no timings for the respawns, but I'd say 2-3 minutes in Patch 2.0 and 5+ minutes in Patch 2.1 should be reasonable.
